src/main/java/EPstorage/BlacklistStorage.java

```diff
@@ -0,0 +1,102 @@
+package EPstorage;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+
+import java.io.*;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+
+/**
+ * class that deals with editing the BlackList storage file.
+ *
+ */
+public class BlacklistStorage {
+
+
+    private ObjectMapper mapper = new ObjectMapper();
+    private File file;
+    private InputStream inputStream;
+    private TypeReference<UserProfile> typeReference = new TypeReference<UserProfile>() {
+    };
+
+    private enum BlacklistKeys { KEYS , MOVIES , ID};
+
+
+    public BlacklistStorage() throws FileNotFoundException {
+        file = new File("EPdata/BlacklistStorage.json");
+        this.inputStream = new FileInputStream(file);
+    }
+
+    /**
+     * load blacklisted movies and keywords from JSON storage.
+     *
+     */
+    public void load() throws IOException {
+
+        try {
+            FileReader reader = new FileReader("EPdata/BlacklistStorage.json");
+            JSONParser jsonParser = new JSONParser();
+            JSONObject jsonObject = (JSONObject) jsonParser.parse(reader);
+            System.out.println("jsonObject");
+            System.out.println(jsonObject.get(BlacklistKeys.KEYS.toString()));
+
+            Gson gson = new Gson();
+            Type type = new TypeToken<ArrayList<String>>() { }.getType();
+            ArrayList<String> keyList = gson.fromJson(jsonObject.get(BlacklistKeys.KEYS.toString()).toString()
+                    , type);
+
+            Type type2 = new TypeToken<ArrayList<MovieModel>>() { }.getType();
+            ArrayList<MovieModel> movieList = gson.fromJson(jsonObject.get(BlacklistKeys.ID.toString()).toString()
+                    , type2);
+
+
+            ArrayList<String> movieTitleList = gson.fromJson(jsonObject.get(BlacklistKeys.MOVIES.toString()).toString()
+                    , type);
+
+
+
+            Blacklist.initialiseAll(keyList , movieTitleList , movieList);
+
+
+
+        } catch (Exception e) {
+            //TODO add exception handling
+        }
+
+    }
+
+    /**
+     * update json file with any changes made blacklist.
+     *
+     * @param blackListKeyWords
+     * @param blackListMovies
+     * @param blackListMoviesTitle
+     */
+    public void updateBlacklistFile (ArrayList<String>  blackListKeyWords ,
+                                    ArrayList<MovieModel>  blackListMovies ,
+                                     ArrayList<String>  blackListMoviesTitle) throws IOException {
```

Comment:
We are following OSS coding standards for this module. Hence, `Indentation for wrapped lines should be 8 spaces`

https://nuscs2113-ay1920s1.github.io/website/coding-standards/java/intermediate.html#layout